src/hotspot/share/cds/filemap.cpp

```diff
@@ -963,7 +1068,19 @@ bool FileMapInfo::validate_app_class_paths(int shared_app_paths_len) {
     int j = header()->app_class_paths_start_index();
     mismatch = check_paths(j, shared_app_paths_len, rp_array);
     if (mismatch) {
-      return classpath_failure("[APP classpath mismatch, actual: -Djava.class.path=", appcp);
+      char* dumptime_prefix = NULL;
+      if (header()->common_app_classpath_size() > 0) {
+        dumptime_prefix = (char*)header() + header()->common_app_classpath_offset();
+      }
+      const char* runtime_prefix = (const char*)longest_common_app_classpath(shared_app_paths_len, rp_array);
+      if (dumptime_prefix != NULL && runtime_prefix != NULL) {
+        mismatch = check_paths_ignoring_common_path(j, shared_app_paths_len, rp_array,
+                                                    (const char*)dumptime_prefix, runtime_prefix);
+      }
```

Comment:
For completeness, we should allow empty prefixes. For example:

dump: `-cp foo.jar:x/bar.jar`
run: `-cp a/foo.jar:a/x/bar.jar`

or

dump: `-cp a/foo.jar:a/x/bar.jar`
run: `-cp foo.jar:x/bar.jar`

Reply:
My updated change doesn't have the condition check anymore.